agent/lib/kontena/workers/stats_worker.rb

```diff
@@ -155,15 +156,88 @@ def send_statsd_metrics(name, event)
       end
       statsd.gauge("#{key_base}.cpu.usage", event[:cpu][:usage_pct])
       statsd.gauge("#{key_base}.memory.usage", event[:memory][:usage])
-      interfaces = event.dig(:network, :interfaces) || []
-      interfaces.each do |iface|
-        [:rx_bytes, :tx_bytes].each do |metric|
-          statsd.gauge("#{key_base}.network.iface.#{iface[:name]}.#{metric}", iface[metric])
-        end
-      end
+      statsd.gauge("#{key_base}.network.internal.rx_bytes", event[:network][:internal][:rx_bytes])
+      statsd.gauge("#{key_base}.network.internal.tx_bytes", event[:network][:internal][:tx_bytes])
+      statsd.gauge("#{key_base}.network.external.rx_bytes", event[:network][:external][:rx_bytes])
+      statsd.gauge("#{key_base}.network.external.tx_bytes", event[:network][:external][:tx_bytes])
     rescue => exc
       error "#{exc.class.name}: #{exc.message}"
       error exc.backtrace.join("\n")
     end
+
+    def calculate_network_traffic(prev_stat, current_stat)
+      prev_interfaces = prev_stat.dig(:network, :interfaces)
+      current_interfaces = current_stat.dig(:network, :interfaces)
+
+      results = {
+        internal: {
+          interfaces: [],
+          rx_bytes: 0,
+          rx_bytes_per_second: 0,
+          tx_bytes: 0,
+          tx_bytes_per_second: 0
+        },
+        external: {
+          interfaces: [],
+          rx_bytes: 0,
+          rx_bytes_per_second: 0,
+          tx_bytes: 0,
+          tx_bytes_per_second: 0
+        }
+      }
+
+      return results unless prev_interfaces and current_interfaces
+
+      prev_timestamp = Time.parse(prev_stat[:timestamp])
+      current_timestamp = Time.parse(current_stat[:timestamp])
+      interval_seconds = current_timestamp - prev_timestamp
+
+      internal_interfaces = current_interfaces.select { |iface|
+        iface[:name].to_s == "ethwe"
+      }
+
+      external_interfaces = current_interfaces.select { |iface|
+        iface[:name].to_s == "eth0"
+      }
+
+      results[:internal] = calculate_interface_traffic(internal_interfaces, prev_interfaces, interval_seconds)
+      results[:external] = calculate_interface_traffic(external_interfaces, prev_interfaces, interval_seconds)
+
+      results
+    end
+
+
+    def calculate_interface_traffic(current_interfaces, prev_interfaces, interval_seconds)
```

Comment:
Agreed.  This and the other refactor fix have been pushed to a new StatsHelper module.